FEXCore/include/FEXCore/Utils/ThreadPoolAllocator.h

```diff
@@ -427,13 +427,20 @@ class FixedSizePooledAllocation final {
   /**
    * @brief Return the owned buffer or allocate another one from the `Allocator`
    *
-   * The buffer returned isn't guaranteed to be the exact size of `Size` but it will be at least `Size`.
-   * The contents of the memory returned isn't guaranteed to be zero initialized or not.
-   * Not even guaranteed to contain the previous data from the previous reowning if the pointer is the same.
+   * The buffer is guaranteed to have at least `Size` bytes of data.
+   * The initial data in the buffer is undefined, even when the buffer is just reowned.
    *
-   * @return object of type `Type` allocated with at least the size of `Size` from the constructor
+   * @param NewSize Optional new size for managed data
+   *
+   * @return object of type `Type` allocated within the selected buffer
    */
-  Type ReownOrClaimBuffer() {
+  Type ReownOrClaimBuffer(std::optional<size_t> NewSize = std::nullopt) {
+    Size = NewSize.value_or(Size);
+    if (NewSize && ClientOwnedFlag != FEXCore::Utils::IntrusivePooledAllocator::ClientFlags::FLAG_FREE && (*Info)->Size < NewSize.value()) {
```

Comment:
@Sonicadvance1 Is reading `(*Info)->Size` thread-safe for claimed-but-not-owned buffers? If not, we would have to move this check past ReownOrClaimBuffer and re-try if we find the size is insufficient.

Reply:
It's not safe.

Reply:
Fixed - we'll now first attempt to reown the buffer and check its size, and then disclaim it and claim a new one if needed.